opm/simulators/linalg/ISTLSolverEbos.hpp

```diff
@@ -520,10 +520,7 @@ namespace Opm
 
             typedef typename Matrix::size_type size_type;
             size_type numCells = grid.size( 0 );
-            blockJacobiForGPUILU0_.reset(new Matrix(numCells, numCells, Matrix::random));
-
-            std::vector<std::set<size_type>> pattern;
-            pattern.resize(numCells);
+            blockJacobiForGPUILU0_.reset(new Matrix(numCells, numCells, 8, 0.4, Matrix::implicit));
```

Comment:
@blattms These number are arbitrarily chosen, based on NORNE and the example in the documentation.
How can we determine better numbers?

Reply:
I must admit that I have previously also struggled with this build mode (in AMG).
After meditating over the documentation of the implicit build mode, I am assuming that the using 0.0 as the first parameter does not work, does it?

Maybe we can at least estimate the 4th factor. We now that no row can have more than 8 entries for 3D CpGrid. If we calculate the average row size of the original matrix as a=ceil((double(A.nnz()/A.N()), we could use cs= 1.05*(8-a). Does that make sense / work?

As a safety net we could catch the exception and use another build mode (row_wise) for the case the estimate was not good enough.

Anyway we should probably also benchmark implicit versus row_wise mode.

Reply:
For NORNE, row_wise is about 2 ms slower. More jacobi blocks means faster building.
For a bigger case, it varied wildly on my machine. Implicit between 1.5 and 12 s, row_wise between 1.2 and 8.5 s.
I think row_wise is still better overall.